.github/workflows/formatting.yaml

```diff
@@ -0,0 +1,16 @@
+name: Check code formatting
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on:
+      - ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: haskell-actions/run-fourmolu@v9
+        with:
+          version: "0.14.0.0"
```

Comment:
That's nice to have the `fourmolu` version pinned here and to have the CI check the formatting :)

Reply:
Maybe we can also have [`nix fmt`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-fmt) set up to format the codebase with `haskellPackages.fourmolu_0_14_0_0` :)

Reply:
We have a nix dev shell, we should make sure this version of `fourmolu` is available there. And then we can just run that. c.f. https://github.com/haskell/lsp/pull/519

Reply:
But I can tweak this later

Reply:
Added fourmolu to the devshell.